pilot/pkg/bootstrap/mesh.go

```diff
@@ -115,6 +118,24 @@ func (s *Server) initMeshNetworks(args *PilotArgs, fileWatcher filewatcher.FileW
 		log.Info("mesh networks configuration not provided")
 		s.environment.NetworksWatcher = mesh.NewFixedNetworksWatcher(nil)
 	}
+	s.environment.AddNetworksHandler(func() {
```

Comment:
My question is we have already registered a network handle, which will trigger xds push

Reply:
Those network handles does not trigger a push when endpoint changes. Only if gateway changes they push.

Reply:
So can we merge them? Currently they are scattered and looks difficult to maintain

Reply:
Yes. I agree. I will do a follow-up and merge them. It needs little refactoring because part of it is in network manager.

Reply:
SGTM